{"ast":null,"code":"import axios from \"axios\";\nimport { toast } from \"react-toastify\";\n\nconst errorCallBack = error => {\n  const expectedError = error.response && error.response.status >= 400 && error.response.status < 500;\n\n  if (!expectedError) {\n    console.log(\"logging the error\", error);\n    toast.error(\"an unexpected error occured\");\n  }\n\n  return Promise.reject(error);\n};\n\naxios.defaults.headers.common[\"x-auth-token\"] = localStorage.getItem(\"token\"); //intercept errors while communicating with backend server\n\naxios.interceptors.response.use(null, errorCallBack);\nexport default {\n  get: axios.get,\n  post: axios.post,\n  put: axios.put,\n  delete: axios.delete\n};","map":{"version":3,"sources":["/Users/prashantkansal/Desktop/techgropse/digiTender/digiTenderFrontend/src/services/httpService.js"],"names":["axios","toast","errorCallBack","error","expectedError","response","status","console","log","Promise","reject","defaults","headers","common","localStorage","getItem","interceptors","use","get","post","put","delete"],"mappings":"AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAASC,KAAT,QAAsB,gBAAtB;;AAEA,MAAMC,aAAa,GAAIC,KAAD,IAAW;AAC/B,QAAMC,aAAa,GACjBD,KAAK,CAACE,QAAN,IACAF,KAAK,CAACE,QAAN,CAAeC,MAAf,IAAyB,GADzB,IAEAH,KAAK,CAACE,QAAN,CAAeC,MAAf,GAAwB,GAH1B;;AAIA,MAAI,CAACF,aAAL,EAAoB;AAClBG,IAAAA,OAAO,CAACC,GAAR,CAAY,mBAAZ,EAAiCL,KAAjC;AACAF,IAAAA,KAAK,CAACE,KAAN,CAAY,6BAAZ;AACD;;AACD,SAAOM,OAAO,CAACC,MAAR,CAAeP,KAAf,CAAP;AACD,CAVD;;AAWAH,KAAK,CAACW,QAAN,CAAeC,OAAf,CAAuBC,MAAvB,CAA8B,cAA9B,IAAgDC,YAAY,CAACC,OAAb,CAAqB,OAArB,CAAhD,C,CACA;;AACAf,KAAK,CAACgB,YAAN,CAAmBX,QAAnB,CAA4BY,GAA5B,CAAgC,IAAhC,EAAsCf,aAAtC;AAEA,eAAe;AACbgB,EAAAA,GAAG,EAAElB,KAAK,CAACkB,GADE;AAEbC,EAAAA,IAAI,EAAEnB,KAAK,CAACmB,IAFC;AAGbC,EAAAA,GAAG,EAAEpB,KAAK,CAACoB,GAHE;AAIbC,EAAAA,MAAM,EAAErB,KAAK,CAACqB;AAJD,CAAf","sourcesContent":["import axios from \"axios\";\nimport { toast } from \"react-toastify\";\n\nconst errorCallBack = (error) => {\n  const expectedError =\n    error.response &&\n    error.response.status >= 400 &&\n    error.response.status < 500;\n  if (!expectedError) {\n    console.log(\"logging the error\", error);\n    toast.error(\"an unexpected error occured\");\n  }\n  return Promise.reject(error);\n};\naxios.defaults.headers.common[\"x-auth-token\"] = localStorage.getItem(\"token\");\n//intercept errors while communicating with backend server\naxios.interceptors.response.use(null, errorCallBack);\n\nexport default {\n  get: axios.get,\n  post: axios.post,\n  put: axios.put,\n  delete: axios.delete,\n};\n"]},"metadata":{},"sourceType":"module"}